src/applications/common/schemaform/save-in-progress/actions.js

```diff
@@ -196,23 +204,32 @@ export function saveAndRedirectToReturnUrl(...args) {
  *                                version of the form the data was saved with
  *                                is different from the current version.
  */
-export function fetchInProgressForm(formId, migrations, prefill = false, prefillTransformer = null) {
+export function fetchInProgressForm(formId, migrations, prefill = false, prefillTransformer = null, prestartForm = null) {
+
   // TODO: Migrations currently aren’t sent; they’re taken from `form` in the
   //  redux store, but form.migrations doesn’t exist (nor should it, really)
-  return (dispatch, getState) => {
+  return async (dispatch, getState) => {
+    if (prestartForm) {
+      try {
+        await prestartForm();
+      // If prestart error, fail safely
+      } catch (err) {
+        return;
+      }
+    }
     const trackingPrefix = getState().form.trackingPrefix;
     const userToken = sessionStorage.userToken;
     // If we don’t have a userToken, fail safely
     if (!userToken) {
       dispatch(setFetchFormStatus(LOAD_STATUSES.noAuth));
-      return Promise.resolve();
+      Promise.resolve();
```

Comment:
Probably need to return the resolved promise or return nothing if you don't want this fetch function to be thennable. `Promise.resolve()` creates a value but doesn't implicitly `return`

Reply:
Good idea, updated.